filesystems/filesystems.tex

```diff
@@ -296,7 +309,7 @@ \subsection{Directory API}
 Any file descriptors opened, and any memory allocated, by opendir are never released.
 This means eventually the process will run out of resources and an \keyword{open} or \keyword{opendir} call will fail.
 
-The fix is to ensure we free up resources in every possible code-path.
+The fix is to ensure we free up resources in every possible code-path (often why kernel error handlers use \keyword{goto}s extensively).
```

Comment:
The real reason for gotos is not to add an additional stack frame. One could write a series of functions and make sure that resources are cleaned up at the end through nesting.

Reply:
Okay, I'll remove it then.